http/client.go

```diff
@@ -290,7 +290,9 @@ func (c *Client) sendRateLimitedRequest(
 		if err != nil {
 			return nil, errors.Newf(err, "failed executing the request %s", URL)
 		}
-		if resp.StatusCode != http.StatusRequestEntityTooLarge || resp.Header.Get("Retry-After") == "" {
+		if (resp.StatusCode != http.StatusRequestEntityTooLarge &&
+			resp.StatusCode != http.StatusServiceUnavailable) ||
```

Comment:
according to the bug, and the openstack-compute API docs (https://docs.openstack.org/mitaka/config-reference/compute/api.html), we should be looking for 403 (http.StatusForbidden), not 503 (http.StatusServiceUnavailable)

(While neither is correct, 503 does seem the more sensible of the two. Are the docs wrong?)